source/user-manual/agents/remote-upgrading/create-custom-wpk-automatically.rst

```diff
@@ -0,0 +1,106 @@
+.. Copyright (C) 2019 Wazuh, Inc.
+
+.. _create-custom-wpk-automatically:
+
+Automated custom WPK packages creation
+============================
+
+Wazuh provides an automated way of building WPK packages.
+
+Requirements
+^^^^^^^^^^^^
+
+    * Docker
+
+1. Get a X509 certificate and CA
+--------------------------------
+
+Create root CA
+^^^^^^^^^^^^^^
+
+.. code-block:: console
+
+    # openssl req -x509 -new -nodes -newkey rsa:2048 -keyout wpk_root.key -out wpk_root.pem -batch
+
+Create a certificate and key
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: console
+
+    # openssl req -new -nodes -newkey rsa:2048 -keyout wpkcert.key -out wpkcert.csr -subj '/C=US/ST=CA/O=Wazuh'
+
+Set the location as follows:
+
+    - /C=US is the country.
+    - /ST=CA is the state.
+    - /O=Wazuh is the organization's name.
+
+Sign this certificate with the root CA
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: console
+
+    # openssl x509 -req -days 365 -in wpkcert.csr -CA wpk_root.pem -CAkey wpk_root.key -out wpkcert.pem -CAcreateserial
+
+2. Build WPK
+--------------------
+
+2.1. Download our wazyh-packages repository from github, there you can find tools for building diferent kinds of packages, and go to the wpk directory.
+
+  .. code-block:: console
+
+    $ git clone https://github.com/wazuh/wazuh-packages && cd wazuh-installers/wpk
+
+2.2. Execute the `generate_wpk_package.sh` script, whith the diferent options you desire. This script will build a Docker image with all the necesary tools to  build the WPK and run a container that will build it:
+
+  .. code-block:: console
+
+    $ ./generate_wpk_package.sh -h
+
+    Usage: /wazuh-installers/wpk-docker/generate_wpk_package.sh [OPTIONS]
+
+    -t,   --target-system <target>              [Required] Select target wpk to build [linux/windows]
+    -b,   --branch <branch>                     [Required] Select Git branch or tag e.g.
+    -d,   --destination <path>                  [Required] Set the destination path of package.
+    -k,   --key-dir <arch>                      [Required] Set the WPK key path to sign package.
+    -a,   --architecture <arch>                 [Optional] Target architecture of the package [x86_64].
+    -j,   --jobs <number>                       [Optional] Number of parallel jobs when compiling.
+    -pd,  --package-directory <directory>       [Required for windows] Path to the package name to pack on wpk.
+    -o,   --output <name>                       [Required] Name to the output package.
+    -h,   --help                                Show this help.
+
+3. Examples
```

Comment:
3. Example title is smaller than 3.1.

Reply:
Title "examples". Why? Why not `wpk generation` or similar?

Reply:
The size of the titles has been corrected and I have changed the example title in https://github.com/wazuh/wazuh-documentation/pull/1308/commits/f1780dd4ad9aa2688d10d80739e742caedb4139e